### PR TITLE
mbed detect: add STM32U5 platforms

### PR DESCRIPTION
**Pull request by jeromecoutant**
### Description

Platforms coming soon:
- NUCLEO_U575ZI_Q
- DISCO_U585I_IOT02A

Regards,

### Pull request type

    [ ] Fix
    [ ] Refactor
    [x] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

**Comment by jeromecoutant** [comment]
@MarceloSalazar

File: src/mbed_os_tools/detect/platform_database.py [hunk 1]
```diff
@@ -177,6 +177,8 @@
         u"0855": u"DISCO_L562QE",
         u"0879": u"NUCLEO_F756ZG",
         u"0885": u"B_L4S5I_IOT01A",
+        u"0886": u"NUCLEO_U575ZI_Q",
+        u"0887": u"DISCO_U585I_IOT02A",
```
Comment (MarceloSalazar) [inline src/mbed_os_tools/detect/platform_database.py:181]: Note this doesn't match with what we have in the internal database:
`DISCO_B_U585I_IOT02A`
What do you prefer us to use?

Comment (jeromecoutant) [reply src/mbed_os_tools/detect/platform_database.py]: I changed to B_U585I_IOT02A :-)

Could you update your database like this ?

Thx

Comment (MarceloSalazar) [reply src/mbed_os_tools/detect/platform_database.py]: Done

**Review by MarceloSalazar — APPROVED** [review]
Board ID and name match with internal database
@ARMmbed/mbed-os-tools please review

**Review by Patater — APPROVED** [review]

**Pull request merged**